std/string.d

```diff
@@ -2114,6 +2114,274 @@ ptrdiff_t lastIndexOfNeither(Char,Char2)(const(Char)[] haystack,
+
+            auto generator = RndValueGen!(
+                GenUnicodeString!(S, 10, 500),
+                GenUnicodeString!(R, 1, 10))
+                    (rnd, ben);
+
+            foreach(S toSearchIn, R toSearchFor; generator)
+            {
+                auto idx = toSearchIn.indexOfNeither(toSearchFor);
+                ben.stop();
+
+                if (idx != -1) {
+                    bool canBeFound;
+                    foreach(dchar it; toSearchFor)
+                    {
+                        if (toSearchIn[idx .. $].front == it)
+                        {
+                            canBeFound = true;
+                            break;
+                        }
+                    }
+
+                    assert(!canBeFound);
+                }
+            }
+        }
+    }
+
+    foreach (S; TypeTuple!(string, wstring, dstring))
+    {
+        foreach(R; TypeTuple!(string, wstring, dstring))
+        {
+            auto ben = Benchmark(format("string.lastIndexOfAny(%s,%s)",
+                S.stringof, R.stringof), rounds);
+
+            auto generator = RndValueGen!(
+                GenUnicodeString!(S, 10, 500),
+                GenUnicodeString!(R, 1, 10))
+                    (rnd, ben);
+
+            foreach(S toSearchIn, R toSearchFor; generator)
+            {
+                auto idx = toSearchIn.indexOfAny(toSearchFor);
+                ben.stop();
+
+                if (idx != -1) {
+                    bool canBeFound;
+                    foreach(dchar it; toSearchFor)
+                    {
+                        if (toSearchIn[idx .. $].front == it)
+                        {
+                            canBeFound = true;
+                            break;
+                        }
+                    }
+
+                    assert(canBeFound);
+                }
+            }
+        }
+    }
+
+    foreach (S; TypeTuple!(string, wstring, dstring))
+    {
+        foreach(R; TypeTuple!(string, wstring, dstring))
+        {
+            auto ben = Benchmark(format("string.lastIndexOfNeither(%s,%s)",
+                S.stringof, R.stringof), rounds);
+
+            auto generator = RndValueGen!(
+                GenUnicodeString!(S, 10, 500),
+                GenUnicodeString!(R, 1, 10))
+                    (rnd, ben);
+
+            foreach(S toSearchIn, R toSearchFor; generator)
+            {
+                auto idx = toSearchIn.indexOfNeither(toSearchFor);
+                ben.stop();
+
+                if (idx != -1) {
+                    bool canBeFound;
+                    foreach(dchar it; toSearchFor)
+                    {
+                        if (toSearchIn[idx .. $].front == it)
+                        {
+                            canBeFound = true;
+                            break;
+                        }
+                    }
+
+                    assert(!canBeFound);
+                }
+            }
+        }
+    }
+}
```

Comment:
This seems to cut against the grain: massive boilerplate of which 90% is identical across different benchmarks, yet impossible to factor away; the actual benchmarked code is lost in the noise.

Ideally benchmark definitions should go together with the functions being benchmarked, in the same manners as unittests. Something like:

```d
auto fun(...)
{
    ... 
}

unittest 
{
   ...
}

auto benchmark_fun(...)
{
    ...
}
```

The challenge here is to have the benchmark framework use introspection and cooperation with the definition of `benchmark_fun` to automate most of the drudgery of selecting arguments, etc. Google's and benchmark framework has benchmark functions take an argument of type `benchmark::State` which includes a variety of things that the benchmark itself gets to use. It seems to me that's a useful model to follow. See https://github.com/google/benchmark. We need considerably less syntactic ugliness because of better introspection.